### ISSUE TITLE
transformer.ipynb : MultiHeadAttention parameter p is never used

### ISSUE THREAD

**Issue by reachtarunhere** [Atcold/pytorch-Deep-Learning#795]
In the `__init__` p is defined but never used. I assume this is some sort of dropout value?

Thanks for the awesome materials :)

```python
class MultiHeadAttention(nn.Module):
    def __init__(self, d_model, num_heads, p, d_input=None):
        super().__init__()
        self.num_heads = num_heads
        self.d_model = d_model
        if d_input is None:
            d_xq = d_xk = d_xv = d_model
        else:
            d_xq, d_xk, d_xv = d_input
            
        # Make sure that the embedding dimension of model is a multiple of number of heads
        assert d_model % self.num_heads == 0

        self.d_k = d_model // self.num_heads
        
        # These are still of dimension d_model. They will be split into number of heads 
        self.W_q = nn.Linear(d_xq, d_model, bias=False)
        self.W_k = nn.Linear(d_xk, d_model, bias=False)
        self.W_v = nn.Linear(d_xv, d_model, bias=False)
        
        # Outputs of all sub-layers need to be of dimension d_model
        self.W_h = nn.Linear(d_model, d_model)
```

**Comment by Atcold** [comment]
Definitely the second version.

A few more edits, since you're at it.
The forward should have two inputs, `x` and `ξ`.
`x` is one input while `ξ` is the secondary one.

`d'` should be the dimensionality of `q` and `k`, while `d''` the dimensionality of `v`.
The matrices should be `h * d` height.
